### PR TITLE
Enhancement: Updated unlinked charts interactions

### PR DESCRIPTION
**Pull request by sohel-rp**
# Description

These changes will,
1. Remove the link chart warning when updating & deploying the unlinked chart
2. Add a subtle message in the left panel
3. Add a new interaction & messaging for manifest output for unlinked charts

Fixes - [AB#1298](https://dev.azure.com/DevtronLabs/Devtron/_workitems/edit/1298)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

All tests have been performed manually by,
- Updating & deploying the external helm app without linking the chart
- Trying to access the manifest output tab & getting the unlinked chart message/tooltip pointing to `connect to helm chart` CTA
- Updating & deploying the external helm app after linking the chart
- Doing some sanity around the Devtron helm app values, the deploy chart, and preset values views

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/v2/values/chartValuesDiff/ChartValuesEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+            value={selectedVersionForDiff}
+            classNamePrefix="compare-values-select"
+            isOptionDisabled={(option) => option.value === 0}
+            formatOptionLabel={formatOptionLabel}
+            components={{
+                IndicatorSeparator: null,
+                ValueContainer: customValueContainer,
+                Option,
+            }}
+            styles={{
```
Comment (vivek-devtron) [inline src/components/v2/values/chartValuesDiff/ChartValuesEditor.tsx:104]: please extract this to some var

File: src/components/v2/values/chartValuesDiff/ChartValuesEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+            label: '',
+            options: [],
+        },
+    ])
+
+    useEffect(() => {
+        const _groupedOptions = []
+        if (deploymentHistoryOptionsList.length > 0) {
+            _groupedOptions.push({
+                label: 'Previous deployments',
```
Comment (vivek-devtron) [inline src/components/v2/values/chartValuesDiff/ChartValuesEditor.tsx:64]: please make constants for these

File: src/components/v2/values/chartValuesDiff/ChartValuesEditor.tsx [hunk 3]
```diff
@@ -0,0 +1,464 @@
...
+
+    useEffect(() => {
+        if (
+            (!comparisonView && valuesForDiffState.selectedVersionForDiff) ||
+            (comparisonView && !valuesForDiffState.selectedVersionForDiff)
+        ) {
+            let _selectedVersionForDiff
+            if (isCreateValueView && selectedChartValues && valuesForDiffState.selectedVersionForDiff) {
+                if (valuesForDiffState.selectedVersionForDiff.value !== selectedChartValues?.id) {
+                    let listToTraverse =
```
Comment (vivek-devtron) [inline src/components/v2/values/chartValuesDiff/ChartValuesEditor.tsx:324]: Please make it const

File: src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx [hunk 4]
```diff
@@ -1095,3 +385,31 @@ export const ErrorScreenWithInfo = ({ info }: { info: string }) => {
         </EmptyState>
     )
 }
+
+export const ConnetToHelmChartTippy = ({
```
Comment (vivek-devtron) [inline src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx:389]: please fix the Typo

Comment (sohel-rp) [reply src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx]: Done

File: src/components/v2/values/chartValuesDiff/ChartValuesEditor.tsx [hunk 5]
```diff
@@ -0,0 +1,464 @@
...
+                    <CodeEditor.Information
+                        className="dc__ellipsis-right"
+                        text="Manifest is generated locally from the YAML."
+                    >
+                        <Tippy
+                            className="default-tt w-250"
+                            arrow={false}
+                            placement="bottom"
+                            content={
+                                'This manifest is generated locally from the YAML. Server-side testing of chart validity (e.g. whether an API is supported) is NOT done. K8s version based templating may be different depending on cluster version.'
```
[the remark this answers is not in the document]
Comment (sohel-rp) [reply src/components/v2/values/chartValuesDiff/ChartValuesEditor.tsx]: Done

File: src/components/v2/values/chartValuesDiff/ChartValuesEditor.tsx (hunk 3 above)
Comment (sohel-rp) [reply src/components/v2/values/chartValuesDiff/ChartValuesEditor.tsx]: Done

File: src/components/v2/values/chartValuesDiff/ChartValuesEditor.tsx (hunk 2 above)
Comment (sohel-rp) [reply src/components/v2/values/chartValuesDiff/ChartValuesEditor.tsx]: Done

File: src/components/v2/values/chartValuesDiff/ChartValuesEditor.tsx (hunk 1 above)
Comment (sohel-rp) [reply src/components/v2/values/chartValuesDiff/ChartValuesEditor.tsx]: Done

File: src/components/v2/values/chartValuesDiff/ChartValuesView.tsx [hunk 6]
```diff
@@ -453,24 +456,40 @@ function ChartValuesView({
...
+
+                        _chartValuesList?.push(_defaultChartValues)
+                        handleChartValuesSelection(_defaultChartValues)
+                    }
+                    setChartValuesList(_chartValuesList)
+                })
+                fetchChartVersionsData(event.chartId, dispatch, commonState.releaseInfo.deployedAppDetail.chartVersion)
+            }
         } else {
+            dispatch({ type: ChartValuesViewActionTypes.repoChartValue, payload: event })
```
Comment (vivek-devtron) [inline src/components/v2/values/chartValuesDiff/ChartValuesView.tsx:492]: The payload is expected to be null or undefined?

File: src/components/v2/values/chartValuesDiff/ChartValuesView.tsx [hunk 7]
```diff
@@ -1231,16 +1275,35 @@ function ChartValuesView({
...
+                        {!isDeployChartView &&
+                            isExternalApp &&
+                            !commonState.installedAppInfo &&
+                            !commonState.showRepoSelector && (
+                                <ConnectToHelmChartTippy
+                                    condition={commonState.showConnectToChartTippy}
+                                    hideConnectToChartTippy={hideConnectToChartTippy}
+                                >
+                                    <InfoColourBar
+                                        message="This app is not connected to a helm chart. Connect to a helm chart to keep up with latest chart versions."
```
Comment (vivek-devtron) [inline src/components/v2/values/chartValuesDiff/ChartValuesView.tsx:1299]: This message should be in const

File: src/components/v2/values/chartValuesDiff/ChartValuesView.constants.ts [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    Label: 'Chart deprecated',
+    InfoText: 'This chart has been deprecated. Please select another chart to continue receiving updates.',
+}
+
+export const CONNECT_CHART_REPO_TEXTS = {
+    InfoText: 'Unable to find the desired chart? To connect or re-sync a repo.',
+    LinkText: 'Go to chart repository',
+}
+
+export const CompareValuesSelectStyles = {
```
Comment (shivani170) [inline src/components/v2/values/chartValuesDiff/ChartValuesView.constants.ts:52]: Why aren't we moving this `CompareValuesSelectStyles` style in  `[ChartRepoSelector.tsx]`selector component

File: src/components/v2/values/chartValuesDiff/ChartValuesView.tsx (hunk 6 above)
Comment (sohel-rp) [reply src/components/v2/values/chartValuesDiff/ChartValuesView.tsx:492]: event check was needed in the case of an external app only.

**Review by vivek-devtron — APPROVED** [review]

File: src/components/v2/values/chartValuesDiff/ChartValuesView.constants.ts (hunk 8 above)
Comment (sohel-rp) [reply src/components/v2/values/chartValuesDiff/ChartValuesView.constants.ts]: Moved to utils where other styles are.

File: src/components/v2/values/chartValuesDiff/ChartValuesView.tsx (hunk 7 above)
Comment (sohel-rp) [reply src/components/v2/values/chartValuesDiff/ChartValuesView.tsx]: Done. Moved all messages to const.

**Review by shivani170 — APPROVED** [review]

**Pull request merged**